File: Python/sentimental-mario-more/mario.py
from cs50 import get_int

def print_row(hashes, spaces, direction):
    if direction == 'l':
        for hashe in range(hashes):
            if hashe < spaces:
                print(" ", end="")
            else:
                print("#", end="")

    elif direction == 'r':
        for hashe in range(0, hashes-spaces, 1):
            print("#", end="")

while True:
    height = get_int("Height: ")

    if height > 0 and height < 9:
        break

for i in range(height):
    spaces = height-i-1
    print_row(height, height-i-1, "l")
    print("  ", end="")
    print_row(height, height-i-1, "r")
    print()

File: C/mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

void renderLineWithGap(int length, int spaces);
void renderLine(int length, int spaces, bool ltr);

int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }while(height<1 || height>8);

    for(int i=0;i<height;i++){
        renderLineWithGap(height,height-1-i);
    }
}

void renderLineWithGap(int length, int spaces)
{
    renderLine(length,spaces,true);
    printf("  ");
    renderLine(length,spaces,false);
    printf("\n");
}

//Prints a line with the char'#'.
void renderLine(int length, int spaces, bool ltr)
{
    for (int i = 0; i < length; i++)
    {
        if(ltr)
        {
            printf("%c", i<spaces ? ' ' : '#');
        }
        else
        {
            if(i<(length-spaces))
            {
                printf("#");
            }
            else
            {
                break;
            }
        }
    }
}
